src/types.ts

```diff
@@ -47,6 +47,34 @@ export interface LegendListRenderItemInfo<ItemT> {
     index: number;
 }
 
+export type LegendListRef = {
```

Comment:
@jmeistrich FlatList and FlashList are defined as class components. Did not found better way then defining separate ref type there.
That's why ref should be defined like this
```
 const scrollViewRef = useRef<LegendListRef>(null);
 ```

Reply:
@mbilenko-florio you can simplify this by using ComponentRef from React to get FlatList's ref type directly:

type LegendListRef = ComponentRef<typeof FlatList>;

This will give you all FlatList's ref methods and keep types in sync automatically.

Reply:
@enestatli Thing here we still want to be compatible, but probably may need some custom API extensions, or API exclusions from this list.

For example FlashList does not support flashScrollIndicators, getScrollRef, etc... We may need to adjust those API's in the future, so having API explicitly defined is bit better at this point.

Reply:
Thanks for the clarification, @mbilenko-florio! That makes sense if future API extensions or exclusions are anticipated. Explicitly defining the API does provide more control and clarity for managing compatibility across components like FlashList.

That said, you could also Pick or Exclude whatever is needed from FlatList's API to make it more solid while still leveraging the existing structure. My suggestion was more about showing how you can directly access FlatList methods using the ComponentRef generic type as a simpler approach.